### ISSUE TITLE
[Question] Why Arquillian Governor is not generic

### ISSUE THREAD

**Issue by lordofthejars** [arquillian/arquillian-governor#36]
Today I was wondering why Governor has one annotation for each provider. For example `@Jira` or `@Github` instead of having `@Issue`

**Comment by smiklosovic** [comment]
Because I have designed it poorly. I would not imagine that you would have so many extensions on top of this so to have it generic was not on the table.

**Comment by smiklosovic** [comment]
There is a lot of boiler plate code which is repeating all over again in every annotation.

Then you have different strategies for resolving issues and so on. It would be required to extract the commonalities. Aint nobody got time for that :)

**Comment by bartoszmajsak** [comment]
Don't be so harsh on yourself @smiklosovic :) You designed well enough at that point of time.

Shall we create an issue to work on it in the future?

**Comment by lordofthejars** [comment]
Yes no worries, I was just sitting down reading the code and the light has opened in my brain. I prefer doing things simple and then abstract them and not on the other way, so I think you approach was the right one.
